src/main/java/net/glowstone/util/PathFinder.java

```diff
@@ -0,0 +1,243 @@
+package net.glowstone.util;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+
+import org.bukkit.util.Vector;
+
+/**
+ * General class for path finding.
+ * TODO: Implement other path finding algorithms (BFS, DFS, D*, potentially more)
+ */
+
+public class PathFinder {
+
+    private final Vector startNode;
+    private final Vector endNode;
+    private List<Vector> path;
+    private Map<Vector, Vector> fromPath;
+
+
+    /**
+     * Creates a new PathFinder object.
+     *
+     * @param startNode Starting position
+     * @param endNode Ending position
+     */
+    public PathFinder(final Vector startNode, final Vector endNode) {
+        // Save our start and end points
+        this.startNode = startNode;
+        this.endNode = endNode;
+        this.path = new ArrayList<Vector>();
```

Comment:
java8, infer generic types plz

Reply:
Fixed - thanks!

Reply:
If you move the initialization of `this.path` up to the declaration, then you can annotate the class `@RequiredArgsConstructor` and Lombok will generate the boilerplate constructor. If you do this, add Javadocs to the `startNode` and `endNode` fields.